webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/assignTaxonomy/BulkAssignTaxonomyModal.js

```diff
@@ -0,0 +1,259 @@
+  fetchLocations,
+  ORGANIZATION_KEY,
+  LOCATION_KEY,
+} from './actions';
+import { foremanUrl } from '../../../../common/helpers';
+import { APIActions } from '../../../../redux/API';
+import {
+  HOSTS_API_PATH,
+  API_REQUEST_KEY,
+} from '../../../../routes/Hosts/constants';
+import TaxonomySelect from './TaxonomySelect';
+
+const BulkAssignTaxonomyModal = ({
+  isOpen,
+  closeModal,
+  selectedCount,
+  fetchBulkParams,
+}) => {
+  const dispatch = useDispatch();
+  const [organizationId, setOrganizationId] = useState('');
+  const [locationId, setLocationId] = useState('');
+  const [orgSelectOpen, setOrgSelectOpen] = useState(false);
+  const [locSelectOpen, setLocSelectOpen] = useState(false);
+  const [orgFixRadioChecked, setOrgFixRadioChecked] = useState(true);
+  const [locFixRadioChecked, setLocFixRadioChecked] = useState(true);
+  const organizations = useSelector(state =>
+    selectAPIResponse(state, ORGANIZATION_KEY)
+  );
+  const organizationStatus = useSelector(state =>
+    selectAPIStatus(state, ORGANIZATION_KEY)
+  );
+  const locations = useSelector(state =>
+    selectAPIResponse(state, LOCATION_KEY)
+  );
+  const locationStatus = useSelector(state =>
+    selectAPIStatus(state, LOCATION_KEY)
+  );
+  const hostUpdateStatus = useSelector(state =>
+    selectAPIStatus(state, BULK_ASSIGN_TAXONOMY_KEY)
+  );
+  const handleModalClose = () => {
+    setOrganizationId('');
+    setLocationId('');
+    setOrgFixRadioChecked(true);
+    setLocFixRadioChecked(true);
+    closeModal();
+  };
+
+  useEffect(() => {
+    dispatch(fetchOrganizations());
+    dispatch(fetchLocations());
+  }, [dispatch]);
+
+  const onOrgToggleClick = () => {
+    setOrgSelectOpen(!orgSelectOpen);
+  };
+  const toggleOrg = toggleRef => (
+    <MenuToggle
+      ref={toggleRef}
+      onClick={onOrgToggleClick}
+      isExpanded={orgSelectOpen}
+      style={{ width: '95%' }}
+    >
+      {getSelectedLabel(organizationId, organizations)}
+    </MenuToggle>
+  );
+
+  const handleOrgSelect = (event, selection) => {
+    setOrganizationId(selection);
+    setOrgSelectOpen(false);
+  };
+
+  const onLocToggleClick = () => {
+    setLocSelectOpen(!locSelectOpen);
+  };
+  const toggleLoc = toggleRef => (
+    <MenuToggle
+      ref={toggleRef}
+      onClick={onLocToggleClick}
+      isExpanded={locSelectOpen}
+      style={{ width: '95%' }}
+    >
+      {getSelectedLabel(locationId, locations)}
+    </MenuToggle>
+  );
+
+  const handleLocSelect = (event, selection) => {
+    setLocationId(selection);
+    setLocSelectOpen(false);
+  };
+
+  const getSelectedLabel = (id, taxonomy) =>
+    taxonomy.results.find(t => t.id === id)?.name;
+
+  const handleError = response => {
+    let message = response.data?.error ?? response.message;
+    if (message.includes('Request failed with status code 422')) {
+      message = __('Update failed because of mismatch in settings');
+    } else if (message.includes('Request failed with status code 400')) {
```

Comment:
Please don't parse error messages. HTTP responses have a code.
```suggestion
    if (response.status == 422) {
      message = __('Update failed because of mismatch in settings');
    } else if (response.status == 400) {
```

Having said that, please make sure the HTTP response contains a proper message you can display instead of coming up with your own. Server side you often know much better why something failed.

Reply:
Somehow the response here is not in that format. There is no `response.status`.

![Screenshot at 2025-06-25 17-16-40](https://github.com/user-attachments/assets/65dfd933-194f-4171-abcc-4d4934036888)

Reply:
Could I have more details about HTTP response? How to set it up and where?

Reply:
`handleError` calls `APIActions.put` which is also in Foreman. I think a good start would be to go where that's defined and do some console logs.

Reply:
Thanks to @ekohl for catching this and @jeremylenz @parthaa for finding out the issue.